Graphs/AbstractGraph.js

```diff
@@ -594,4 +598,103 @@ export default class AbstractGraph extends React.Component {
         return null;
     }
 
+    getGraphDimension = () => {
+        const {
+            height,
+            width,
+            data
+        } = this.props;
+
+        const { legend } = this.getConfiguredProperties();
+
+        let dimensions = {
+            graphWidth: width,
+            graphHeight: height,
+            legendHeight: 0,
+            legendWidth: 0
+        }
+
+        if (!legend.show || data.length <= 1 || !legend.separate) {
+            return dimensions;
+        }
+
+        // Compute the available space considering a legend
+        if (legend.orientation === 'vertical') {
+            const value = this.getValueFromPercentage(width, legend.separate);
+            dimensions = {
+                ...dimensions,
+                graphWidth: width - value,
+                legendWidth: value,
+                legendHeight: height,
+            }
+        } else {
+            const value = this.getValueFromPercentage(height, legend.separate);
+            dimensions = {
+                ...dimensions,
+                graphHeight: height - value,
+                legendHeight: value,
+                legendWidth: width,
+            }
+        }
+
+        return dimensions;
+    }
+
+    getValueFromPercentage = (value, percentage) => {
+        return (percentage * value) / 100;
+    }
+
+    renderSeparateLegend(data, legend, getColor, label, isVertical) {
+
+        let legendStyle = {};
+        if (isVertical) {
+            // Place the legends in the bottom left corner
+            legendStyle = { alignSelf: 'flex-end' }
+        } else {
+            // Place legends horizontally
+            legendStyle = {
+                width: '100%',
+                display: 'grid',
+                gridTemplateColumns: `repeat(auto-fit, minmax(${legend.width}px, 1fr))`,
+            }
+        }
+
+        return (
+            <div className='legend' style={legendStyle}>
+                {this.getLegendContent(data, legend, getColor, label)}
+            </div>
+        );
+    }
+
+    getLegendContent(data, legend, getColor, label) {
```

Comment:
to keep it consistent can we use the same function to render the legend above?

Reply:
@natabal Actually this need to be done, since now it mostly div based and we will be able to add scroll with div based inner HTML, otherwise we need to calculate the height of all the legends and then need to set the width for parent SVG using some calculations.

Reply:
@nxanil What I meant here is to use getLengendContent function above